source/common/ssl/connection_impl.cc

```diff
@@ -187,6 +188,39 @@ std::string ConnectionImpl::sha256PeerCertificateDigest() {
   return Hex::encode(computed_hash);
 }
 
+std::string ConnectionImpl::uriSanPeerCertificate() {
```

Comment:
@mattklein123 
More tests added. connection_impl.cc is now 99% covered. I believe the new function is full covered.